jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codemodernizer/CodeWhispererCodeModernizerGumbyClientTest.kt

```diff
@@ -0,0 +1,176 @@
+            verifyNoInteractions(streamingBearerClient)
+            assertThat(actual).isInstanceOf(CreateUploadUrlResponse::class.java)
+            assertThat(actual).usingRecursiveComparison().comparingOnlyFields("uploadUrl", "uploadId", "kmsKeyArn")
+                .isEqualTo(exampleCreateUploadUrlResponse)
+        }
+    }
+
+    @Test
+    fun `check getCodeModernizationJob`() {
+        val actual = gumbyClient.getCodeModernizationJob("jobId")
+        argumentCaptor<GetTransformationRequest>().apply {
+            verify(bearerClient).getTransformation(capture())
+            verifyNoInteractions(streamingBearerClient)
+            assertThat(actual).isInstanceOf(GetTransformationResponse::class.java)
+            assertThat(actual).usingRecursiveComparison().comparingOnlyFields("jobId", "status", "transformationType", "source")
+                .isEqualTo(exampleGetCodeMigrationResponse)
+        }
+    }
+
+    @Test
+    fun `check startCodeModernization`() {
+        val actual = gumbyClient.startCodeModernization("jobId", TransformationLanguage.JAVA_8, TransformationLanguage.JAVA_17)
+        argumentCaptor<StartTransformationRequest>().apply {
+            verify(bearerClient).startTransformation(capture())
+            verifyNoInteractions(streamingBearerClient)
+            assertThat(actual).isInstanceOf(StartTransformationResponse::class.java)
+            assertThat(actual).usingRecursiveComparison().comparingOnlyFields("transformationJobId").isEqualTo(exampleStartCodeMigrationResponse)
+        }
+    }
+
+    @Test
+    fun `check getCodeModernizationPlan`() {
+        val actual = gumbyClient.getCodeModernizationPlan(JobId("JobId"))
+        argumentCaptor<GetTransformationPlanRequest>().apply {
+            verify(bearerClient).getTransformationPlan(capture())
+            verifyNoInteractions(streamingBearerClient)
+            assertThat(actual).isInstanceOf(GetTransformationPlanResponse::class.java)
+            assertThat(actual).usingRecursiveComparison().comparingOnlyFields("transformationSteps").isEqualTo(exampleGetCodeMigrationPlanResponse)
+        }
+    }
+
+    @Test
+    fun `check stopTransformation`() {
+        val actual = gumbyClient.stopTransformation("JobId")
+        argumentCaptor<StopTransformationRequest>().apply {
+            verify(bearerClient).stopTransformation(capture())
+            verifyNoInteractions(streamingBearerClient)
+            assertThat(actual).isInstanceOf(StopTransformationResponse::class.java)
+            assertThat(actual).usingRecursiveComparison().comparingOnlyFields("transformationStatus").isEqualTo(exampleStopTransformationResponse)
+        }
+    }
+
+    @Test
+    fun `check downloadExportResultArchive`() {
+        val requestCaptor = ArgumentCaptor.forClass(ExportResultArchiveRequest::class.java)
+        val handlerCaptor = ArgumentCaptor.forClass(ExportResultArchiveResponseHandler::class.java)
+        projectCoroutineScope(project).launch {
```

Comment:
Should we use have a mock jobId that is passed around to all these functions? A constant in our test util file or something?

Reply:
make sense. I will change to use the jobId from the test util.